vue-app/src/views/FactoryRound.vue

```diff
@@ -0,0 +1,771 @@
+<template>
+  <div id="app" class="wrapper">
+    <nav-bar :inApp="isInApp" />
+    <div id="content-container">
+      <div
+        id="sidebar"
+        v-if="isSidebarShown"
+        :class="`${$store.state.showCartPanel ? 'desktop-l' : 'desktop'}`"
+      >
+        <round-information />
+      </div>
+      <div
+        id="content"
+        :class="{
+          padded: isSidebarShown && !isCartPadding,
+          'mr-cart-open': isCartToggledOpen && isSideCartShown,
+          'mr-cart-closed': !isCartToggledOpen && isSideCartShown,
+        }"
+      >
+        <back-link
+          v-if="showBackLink"
+          :to="backLinkRoute"
+          :text="backLinkText"
+        />
+        <router-view :key="$route.path" />
+      </div>
+      <div
+        id="cart"
+        v-if="isSideCartShown && !isHistoricRound"
+        :class="`desktop ${isCartToggledOpen ? 'open-cart' : 'closed-cart'}`"
+      >
+        <cart-widget />
+      </div>
+    </div>
+    <mobile-tabs v-if="isMobileTabsShown" />
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Watch } from 'vue-property-decorator'
+
+import { getCurrentRound } from '@/api/round'
+import { User } from '@/api/user'
+
+import RoundInformation from '@/views/RoundInformation.vue'
+import NavBar from '@/components/NavBar.vue'
+import CartWidget from '@/components/CartWidget.vue'
+import MobileTabs from '@/components/MobileTabs.vue'
+import BackLink from '@/components/BackLink.vue'
+
+import {
+  LOAD_USER_INFO,
+  LOAD_ROUND_INFO,
+  LOAD_RECIPIENT_REGISTRY_INFO,
+  SELECT_ROUND,
+  LOAD_CART,
+  LOAD_COMMITTED_CART,
+  LOAD_CONTRIBUTOR_DATA,
+  LOGIN_USER,
+} from '@/store/action-types'
+import {
+  SET_CURRENT_USER,
+  SET_ACTIVE_ROUND_ADDRESS,
+  TOGGLE_SHOW_CART_PANEL,
+} from '@/store/mutation-types'
+
+@Component({
+  name: 'clr.fund',
+  metaInfo: {
+    title: 'clr.fund',
+    titleTemplate: 'clr.fund - %s',
+    meta: [
+      {
+        name: 'git-commit',
+        content: process.env.VUE_APP_GIT_COMMIT || '',
+      },
+    ],
+  },
+  components: {
+    RoundInformation,
+    NavBar,
+    MobileTabs,
+    CartWidget,
+    BackLink,
+  },
+})
+export default class FactoryRound extends Vue {
+  pollingInterval
+
+  async created(): Promise<void> {
+    const factoryAddress =
+      this.$route.params.factoryAddress ||
+      process.env.VUE_APP_CLRFUND_FACTORY_ADDRESS
+    const roundIndex = this.$route.params.roundIndex
+
+    if (!factoryAddress) {
+      throw new Error('No factory address found')
+    }
+
+    // Always fetch the current round and store it. We use it to compare with the
```

Comment:
I feel like we should be consistent about terminology here.

> Always fetch the current round and store it. We use it to compare with the current round that the user might select manually

Here was say "current round" twice but I think you mean two separate things
1) the current round according to the contract - i.e. the latest round
2) the current round the user has selected - i.e. the selected round

You also mention "active round" in `SET_ACTIVE_ROUND_ADDRESS`, which I believe is the same a 1).

Perhaps we sick with "current round" to represent the latest round, given the contract function is `getCurrentRound`. Then we have a concept of a "selected round" that a user may navigate to. Or I'm also ok with "latest round" or "active round" to represent the latest round & "current round" that a user may manually select or navigate to.

Reply:
Yes, this is wrong. I meant to say:

> Always fetch the active round and store it. (or) Always fetch the current round from the contracts and store it as the active round.

Will fix it.

Regarding the terms, I'm using:
- `currentRound`: the pointer used within the app (this is the selected round you mention)
- `activeRound`: this one never change (or not in a round cycle, will change when a new round is deployed)

I liked your suggestion and I think that would make more sense but I wouldn't change the `currentRound` terminology or meaning because it is used a lot and is super embedded in the app. Changing it would mean changing a lot of code.